virtualmachine/vsphere/vm.go

```diff
@@ -1587,3 +1587,36 @@ func getVirtualMachines(vm *VM) ([]mo.VirtualMachine, error) {
 	}
 	return vmList, nil
 }
+
+// ConvertToTemplate : converts vm to vm template
```

Comment:
[Queries] I assume vm-template will keep the same name as vm. Please confirm.
Is there a way to rename vm or vm-template later?

Reply:
Yes, the name of the template is the same as the vm-name. 
Govmomi supports renaming the vm, but do we need it ?